### ISSUE TITLE
Why is SRD0018 not triggered here?

### ISSUE THREAD

**Issue by niphlod** [ErikEJ/SqlServer.Rules#740]
Continuing from #720 .

SRD0018 is "UPDATE statement without row limiting conditions."

with the usual "preparation" of the target table

```
CREATE TABLE dbo.test (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);
```

This is triggered for 

```
DECLARE @someTable TABLE (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN @someTable bodyItems
ON bodyItems.originalId = dst.originalId;
```

but not for 

```
CREATE TABLE #test (
    originalId VARCHAR(50),
    currentId VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN #test bodyItems
ON bodyItems.originalId = dst.originalId;
```

removing the temp table creation re-triggers SRD0018.

This leads to another funny thing happening ..... a very easy way for me to know something is "funky" is adding a ` something <> '';` to trigger `SRP0006: Try to avoid using not equal operator (<>,!=) in the WHERE clause if possible. (Sargable)` .

well....

```
UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN #test bodyItems
ON bodyItems.originalId = dst.originalId
WHERE bodyItems.originalId <> '';
```
raises SRP0006, but adding temp table creation at the top removes the SRP0006, so I *think* the temp table statement breaks the parser.

**Comment by ErikEJ** [comment]
@niphlod SRD0018 is about table variables, not temp tables, so this is expected.
